src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs

```diff
@@ -1090,11 +1247,19 @@ public void TestNestedGenericInheritanceWithMultipleParameters()
             Assert.Equal(NullabilityState.NotNull, item3Info.ElementType.WriteState);
         }
 
-        [Fact]
+        public static IEnumerable<object[]> TestNullabilityInfoCreationOnPropertiesWithNestedGenericTypeArgumentsData() => new[]
+        {
+            new object[] { typeof(bool), NullabilityState.NotNull },
+            new object[] { typeof(bool?), NullabilityState.Nullable },
+            new object[] { typeof(object), NullabilityState.Nullable },
+        };
+
+        [Theory]
         [SkipOnMono("Nullability attributes trimmed on Mono")]
-        public void TestNullabilityInfoCreationOnPropertiesWithNestedGenericTypeArguments()
+        [MemberData(nameof(TestNullabilityInfoCreationOnPropertiesWithNestedGenericTypeArgumentsData))]
+        public void TestNullabilityInfoCreationOnPropertiesWithNestedGenericTypeArguments(Type genericArgumentType, NullabilityState expectedGenericArgumentNullability)
         {
-            Type type = typeof(TypeWithPropertiesNestingItsGenericTypeArgument<int>);
```

Comment:
Previously, this test was using `int` as the generic argument but was reporting back `Nullable` for the state of `T`. That seems incorrect and not consistent with other behavior. Now we report `Nullable` for `object` and `bool?` but `NotNull` for `bool`.

Reply:
Right, thanks